poco: add checkPhase

dwt opened the pull request:
This executes at least part of the testsutie.

All testsuites that require a service running in the background (Postgress, MongoDB, MariaDB, ...) have been skipped. If one attempts to run them, this likely requires a nixosTest to do so and provide the dependencies.

Some tests I had to skip, because they break on darwin. Perhaps fixes for theste test can be developed and upstreamed, but that requires more expertise than I possess.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

dwt committed:
poco: add checkPhase

This executes at least part of the testsutie.

All testsuites that require a service running in the background
(Postgress, MongoDB, MariaDB, ...) have been skipped. If one attempts to
run them, this likely requires a nixosTest to do so and provide the
dependencies.

Some tests I had to skip, because they break on darwin. Perhaps fixes
for theste test can be developed and upstreamed, but that requires more
expertise than I possess.

afh commented on pkgs/by-name/po/poco/disable-broken-tests.patch:
Wondering if there is any benefit in using a `-U0` patch here…

dwt replied:
good idea

afh commented on pkgs/by-name/po/poco/package.nix:
```diff
@@ -67,8 +80,16 @@ stdenv.mkDerivation rec {
       hash = "sha256-VFWuRuf0GPYFp43WKI8utl+agP+7a5biLg7m64EMnVo=";
     })
+    # failing on darwin, could perhaps be patched / a fix upstreamed later
+    ./disable-broken-tests.patch
```
If the tests only fail on darwin, maybe they should be patched out only darwin.
What do you think?

dwt replied:
To be honest, I wasn't able to check on anything but darwin yet. The code of most of these failures look like they probably are also a failure on linux though.

dwt replied:
I haven't seen many conditionally applied patches for tests yet, outside of the largest ports like LibreOffice, and even there only for versions / variants, not for Darwin. Is this something that is usually done in nixpkgs that I haven't yet discovered?

afh replied:
From the top of my head I cannot recall on which packages this happened, yet I was asked to apply patches conditionally if they targeted a specific platform.

dwt replied:
ok. I'll give linux build a try and see if the patches are also required.

afh reviewed (commented):
Thanks for putting in the work, @dwt, please find below a few comments after a first quick. glance.

dwt committed:
poco: seperately disable broken tests on linux and darwin

poco: Better optionals

dwt commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 390150 --skip-package-regex collabora-online`

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 13 packages built:</summary>
  <ul>
    <li>craftos-pc</li>
    <li>ioq3-scion</li>
    <li>ioquake3</li>
    <li>litebrowser</li>
    <li>mumble</li>
    <li>murmur</li>
    <li>poco</li>
    <li>poco.dev</li>
    <li>pothos</li>
    <li>projectm-sdl-cpp</li>
    <li>quake3demo</li>
    <li>sanjuuni</li>
    <li>vscode-extensions.jackmacwindows.craftos-pc</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:fast_forward: 2 packages marked as broken and skipped:</summary>
  <ul>
    <li>litebrowser</li>
    <li>sanjuuni</li>
  </ul>
</details>
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>projectm-sdl-cpp</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>poco</li>
    <li>poco.dev</li>
  </ul>
</details>

dwt commented:
I think this pul request is about ready to be merged, therefore I would like to request a review from @trofi, @jnsgruk, @tomodachi94 and @afh. Many thanks in advance

dwt commented:
Some open questions, that I would love some input on:

- There can be more effort to get more of the test suite running (for example via `nixosTests`. My current thinking is, that it is way better to get some tests running than to get none running
- Darwin does not have a sandbox in the same way than linux does, this leads to a different set of tests failing on both platforms. My understanding is, that there has recently been a breakthrough in understanding how the nix sandbox can be enabled on darwin, but is not yet merged. Since I do not know when this will land, I wanted to get started with the current set of failing tests. However, should I just be disabling the superset of failing tests on both platforms and be done with it, or is it preferred to to have this separated as it is now?

@afh: Regarding `projectm-sdl-cpp` - yes, quite possibly, I wouldn't want to burden this pull request with that. I would also be glad if someone else did this.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/5300

dwt commented:
I've added a pull request for [projectm-sdl-cpp](https://github.com/NixOS/nixpkgs/pull/392047)  to mark it as not compiling on darwin.

dwt commented:
@orivej @tomodachi94 do you have any thoughts about this change to the package you are maintaining?

dwt commented:
Friendly Ping @tomodachi94 What do you think of these changes? Would you need anything else to merge them?

dwt commented:
Freindly ping to @orivej and @tomodachi94 - are you still active on nixpkgs?

drupol commented on pkgs/by-name/po/poco/package.nix:
```diff
@@ -54,20 +54,45 @@ stdenv.mkDerivation rec {
+        hash = "sha256-VFWuRuf0GPYFp43WKI8utl+agP+7a5biLg7m64EMnVo=";
+      })
+    ]
+    ++ lib.optional stdenv.hostPlatform.isDarwin ./disable-broken-tests-darwin.patch
+    ++ lib.optional stdenv.hostPlatform.isLinux ./disable-broken-tests-linux.patch;
```
```suggestion
    ++ lib.optionals stdenv.hostPlatform.isDarwin [
    ./disable-broken-tests-darwin.patch
  ] ++ lib.optionals stdenv.hostPlatform.isLinux [
    ./disable-broken-tests-linux.patch;
  ];
```

dwt replied:
Sure fine, but could you perhaps give some rationale why this is better? Is this about the shorter line length? It seems that there is no real difference between `lib.optionals()` vs. `lib.optional()`?

drupol replied:
The brackets are explicit, it makes things easier to reason about, especially for newcomers in the nix language.

drupol commented on pkgs/by-name/po/poco/package.nix:
```diff
@@ -54,20 +54,45 @@ stdenv.mkDerivation rec {
+  doCheck = true;
+  preCheck = ''
+    # workaround for some tests trying to write to /homeless-shelter
+    export HOME=$(mktemp -d)
```
Remove this and use `writableTmpDirAsHomeHook`

dwt committed:
poco: switch to writableTmpDirAsHomeHook for a writeable home directory during tests

dwt committed:
poco: prevent warning for --replace without -fail (and other suffixes)

drupol commented:
There's no need to have a commit for the changes you made with optionals. Please merge it in the commit where you added it initially

drupol commented:
The commit https://github.com/NixOS/nixpkgs/pull/390150/commits/d1446fd04da79f5fcd9c8e955645acd73dd1b4a5 is incorrectly formatted. Please adjust.

dwt commented:
@drupol 

> The commit [d1446fd](https://github.com/NixOS/nixpkgs/commit/d1446fd04da79f5fcd9c8e955645acd73dd1b4a5) is incorrectly formatted. Please adjust.

Could you please give me a hint what I did wrong? To my understanding neither the formatting check in nixpkgs triggered, nor did I change the formatting of that line in the commit? I also have auto formatting with nixpkgs format set in my editor, so I'm surprised I missed this?

drupol commented:
Please make sure they are consistently in the pattern. The two first commits are correct, the last one is not.

Have a look at pretty much all the commits in nixpkgs to have an idea. 

It hasn't been reported by the ci because there's no automated checks for this, this is the reason why we do pull requests reviews manually.

dwt commented:
🤦Ah.... you mean the commit message. [x] fixed.

drupol commented on pkgs/by-name/po/poco/package.nix:
```diff
@@ -54,25 +55,54 @@ stdenv.mkDerivation rec {
+  patches =
+    [
+      # Remove on next release
+      (fetchpatch {
```
optional: I have seen on some PR that `fetchpatch2` should now be used.

dwt replied:
[That's still so new, it's not yet documented](https://nixos.org/manual/nixpkgs/unstable/#fetchpatch). The only difference seems to be, that it uses a newer version of patchutils, to combat aha-breakage on GitHub. The Code does not make it clear when this might happen, but none the less that seems like a good idea.

dwt replied:
Yeah, also the fetchpatch was not actually added in this pull request, but merely a whitespace change. I kind of hope that once the automated change to fetchpatch2 is through, there will once again be only a fetchpatch and no two of them, as that is uneccessarily confusing.

drupol commented on pkgs/by-name/po/poco/package.nix:
```diff
@@ -54,25 +55,54 @@ stdenv.mkDerivation rec {
     grep -rlF INTERFACE_INCLUDE_DIRECTORIES "$dev/lib/cmake/Poco" | while read -r f; do
       substituteInPlace "$f" \
-        --replace "$"'{_IMPORT_PREFIX}/include' ""
+        --replace-quiet "$"'{_IMPORT_PREFIX}/include' ""
```
Why not `--replace-fail` ?

dwt replied:
I tried, but the way the replace is phrased, it tries the replace on many files that do not contain the replace string. I thought it better to at least get rid of the warning this way.

drupol replied:
Perhaps we could identify those files and filter them? I don't think using replace-quiet is a good practice.

dwt replied:
:/ I'm not loving it.

Something like this would be possible

```
grep -rlF INTERFACE_INCLUDE_DIRECTORIES "$dev/lib/cmake/Poco" | \
while read -r f; do
  if grep -qF '${_IMPORT_PREFIX}/include' "$f"; then
    substituteInPlace "$f" \
      --replace-fail '$'"{_IMPORT_PREFIX}/include" ""
  fi
done
```

But to me, I don't really see what its adding over `--replace-quiet`

dwt replied:
Sadly the original commit the added this, doesn't really say what and why this is necessary, so I'm kind of lost as to what this is doing and why it is necessary.

dwt commented:
> Do you think you could fix the broken build on aarch64-darwin ?

@drupol Do you mean of `projectm-sdl-cpp` or do you mean something different?

drupol commented:
See https://logs.ofborg.org/?key=nixos/nixpkgs.390150&attempt_id=d742aead-8f58-4db2-a197-83fa1f165c20

dwt commented:
> See https://logs.ofborg.org/?key=nixos/nixpkgs.390150&attempt_id=d742aead-8f58-4db2-a197-83fa1f165c20

That is quite strange, I cannot reproduce this. :-( @drupol can you? Or do you have an idea how I could?

I could disable the tests that failed only on ofborg, is that the recommended way in nixpkgs? I would love some guidance here.

drupol commented:
Can you rebase against `master` and try again?

drupol commented:
@GaetanLepage Could you trigger a `nixpkgs-review` build on this PR ?

afh commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 390150 --package poco`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>poco</li>
    <li>poco.dev (poco.dev.dev)</li>
  </ul>
</details>

GaetanLepage commented:
Can you please rebase on `master` and force-push?

GaetanLepage commented:
The last commit should be squashed in the first one.

GaetanLepage commented:
> @GaetanLepage While I'm not against that, the two commits do not seem to be for the same topic? Adding the check-phase disjunct from fixing a warning that was already exiting? Am I missing something?

I have no strong feeling, do as you want. I suggested this as such small changes can be grouped in "bigger" commits.
Anyway, this is clearly suggestive and your commit history is perfectly valid.

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 390150`
Commit: `45a6393b41f60be08d020765b667f577d91d4b2f`

---
### `x86_64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>quake3arena</li>
    <li>quake3arena-hires</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 16 packages built:</summary>
  <ul>
    <li>collabora-online</li>
    <li>craftos-pc</li>
    <li>ioq3-scion</li>
    <li>ioquake3</li>
    <li>multipass</li>
    <li>mumble</li>
    <li>mumble_overlay</li>
    <li>murmur</li>
    <li>poco</li>
    <li>poco.dev</li>
    <li>pothos</li>
    <li>projectm-sdl-cpp</li>
    <li>quake3demo</li>
    <li>quake3demo-hires</li>
    <li>sanjuuni</li>
    <li>vscode-extensions.jackmacwindows.craftos-pc</li>
  </ul>
</details>

dwt committed:
poco: disable flaky tests

These tests have proven flaky in the nixpkgs ci.

Likely because nix build boxes are constantly under very high load, and
thus take longer than expected for sleep calls, which is what all of
these tests try to do.

GaetanLepage reviewed (approved):
Yes, LGTM

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 390150`

Logs: https://github.com/GaetanLepage/nixpkgs-review-gha/actions/runs/16252373079


---
### `x86_64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>quake3arena</li>
    <li>quake3arena-hires</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 16 packages built:</summary>
  <ul>
    <li>collabora-online</li>
    <li>craftos-pc</li>
    <li>ioq3-scion</li>
    <li>ioquake3</li>
    <li>multipass</li>
    <li>mumble</li>
    <li>mumble_overlay</li>
    <li>murmur</li>
    <li>poco</li>
    <li>poco.dev</li>
    <li>pothos</li>
    <li>projectm-sdl-cpp</li>
    <li>quake3demo</li>
    <li>quake3demo-hires</li>
    <li>sanjuuni</li>
    <li>vscode-extensions.jackmacwindows.craftos-pc</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>quake3arena</li>
    <li>quake3arena-hires</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 14 packages built:</summary>
  <ul>
    <li>collabora-online</li>
    <li>craftos-pc</li>
    <li>ioq3-scion</li>
    <li>ioquake3</li>
    <li>mumble</li>
    <li>murmur</li>
    <li>poco</li>
    <li>poco.dev</li>
    <li>pothos</li>
    <li>projectm-sdl-cpp</li>
    <li>quake3demo</li>
    <li>quake3demo-hires</li>
    <li>sanjuuni</li>
    <li>vscode-extensions.jackmacwindows.craftos-pc</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 9 packages failed to build:</summary>
  <ul>
    <li>ioq3-scion</li>
    <li>ioquake3</li>
    <li>mumble</li>
    <li>poco</li>
    <li>poco.dev</li>
    <li>quake3arena</li>
    <li>quake3arena-hires</li>
    <li>quake3demo</li>
    <li>quake3demo-hires</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>poco</summary>
<pre>testIndex: 

!!!FAILURES!!!
Runs: 839   Failures: 1   Errors: 0

There was 1 failure: 
 1: CppUnit::TestCaller<UniqueExpireLRUCacheTest>.testExpireN
    "aCache.has(3)"
    in "/nix/build/nix-build-poco-1.14.1.drv-0/source/Foundation/testsuite/src/UniqueExpireLRUCacheTest.cpp", line 135



92% tests passed, 1 tests failed out of 12

Total Test time (real) = 290.91 sec

The following tests FAILED:
	  1 - Foundation (Failed)
Errors while running CTest
make: *** [Makefile:91: test] Error 8</pre>
</details>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 9 packages failed to build:</summary>
  <ul>
    <li>ioq3-scion</li>
    <li>ioquake3</li>
    <li>mumble</li>
    <li>poco</li>
    <li>poco.dev</li>
    <li>quake3arena</li>
    <li>quake3arena-hires</li>
    <li>quake3demo</li>
    <li>quake3demo-hires</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>poco</summary>
<pre>testIndex: 

!!!FAILURES!!!
Runs: 839   Failures: 1   Errors: 0

There was 1 failure: 
 1: CppUnit::TestCaller<UniqueExpireLRUCacheTest>.testExpireN
    "aCache.has(1)"
    in "/nix/build/nix-build-poco-1.14.1.drv-0/source/Foundation/testsuite/src/UniqueExpireLRUCacheTest.cpp", line 126



92% tests passed, 1 tests failed out of 12

Total Test time (real) = 260.48 sec

The following tests FAILED:
	  1 - Foundation (Failed)
Errors while running CTest
make: *** [Makefile:91: test] Error 8</pre>
</details>
</details>

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 390150`

Logs: https://github.com/GaetanLepage/nixpkgs-review-gha/actions/runs/16252612406


---
### `x86_64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>quake3arena</li>
    <li>quake3arena-hires</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 16 packages built:</summary>
  <ul>
    <li>collabora-online</li>
    <li>craftos-pc</li>
    <li>ioq3-scion</li>
    <li>ioquake3</li>
    <li>multipass</li>
    <li>mumble</li>
    <li>mumble_overlay</li>
    <li>murmur</li>
    <li>poco</li>
    <li>poco.dev</li>
    <li>pothos</li>
    <li>projectm-sdl-cpp</li>
    <li>quake3demo</li>
    <li>quake3demo-hires</li>
    <li>sanjuuni</li>
    <li>vscode-extensions.jackmacwindows.craftos-pc</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>quake3arena</li>
    <li>quake3arena-hires</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 14 packages built:</summary>
  <ul>
    <li>collabora-online</li>
    <li>craftos-pc</li>
    <li>ioq3-scion</li>
    <li>ioquake3</li>
    <li>mumble</li>
    <li>murmur</li>
    <li>poco</li>
    <li>poco.dev</li>
    <li>pothos</li>
    <li>projectm-sdl-cpp</li>
    <li>quake3demo</li>
    <li>quake3demo-hires</li>
    <li>sanjuuni</li>
    <li>vscode-extensions.jackmacwindows.craftos-pc</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 7 packages failed to build:</summary>
  <ul>
    <li>ioq3-scion</li>
    <li>ioquake3</li>
    <li>mumble</li>
    <li>quake3arena</li>
    <li>quake3arena-hires</li>
    <li>quake3demo</li>
    <li>quake3demo-hires</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>poco</li>
    <li>poco.dev</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 7 packages failed to build:</summary>
  <ul>
    <li>ioq3-scion</li>
    <li>ioquake3</li>
    <li>mumble</li>
    <li>quake3arena</li>
    <li>quake3arena-hires</li>
    <li>quake3demo</li>
    <li>quake3demo-hires</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>poco</li>
    <li>poco.dev</li>
  </ul>
</details>

dwt commented:
@GaetanLepage I don't quite understand why so many packages fail to build for you. When I run the build locally, I only get build failures with ioq3-scion, which is quite finicky on Mac anyway (not sure it ever built, I tried to contribute patches but I haven't kept up to date with that package).

Looking at [what I think is the build log from your GitHub action](https://github.com/GaetanLepage/nixpkgs-review-gha/actions/runs/16252612406/job/45884336732) I also don't understand why the other packages besides poco have failed there. My reeding of that log seems to suggest that nix didn't even try to build them?

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 390150`
Commit: `2a8e772fb192678b22c7afba4eb4851fa9a3d3a2`

---
### `aarch64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>ioq3-scion</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>ioquake3</li>
    <li>mumble</li>
    <li>poco</li>
    <li>poco.dev</li>
    <li>quake3arena</li>
    <li>quake3arena-hires</li>
    <li>quake3demo</li>
    <li>quake3demo-hires</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>ioq3-scion</summary>
<pre>      _Netchan_InitSession in net_chan.o
  "_sodium_base642bin", referenced from:
      _Netchan_InitSession in net_chan.o
  "_sodium_bin2base64", referenced from:
      _CL_CheckForResend in cl_main.o
      _SV_Init in sv_init.o
  "_sodium_increment", referenced from:
      _Netchan_SendEncrypted in net_chan.o
  "_sodium_init", referenced from:
      _main in sys_main.o
  "_sodium_memzero", referenced from:
      _CL_ConnectionlessPacket in cl_main.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [Makefile:2447: build/release-darwin-arm64/ioquake3.arm64] Fehler 1
rm build/release-darwin-arm64/renderergl2/glsl/bokeh_fp.c build/release-darwin-arm64/renderergl2/glsl/calclevels4x_fp.c build/release-darwin-arm64/renderergl2/glsl/down4x_fp.c build/release-darwin-arm64/renderergl2/glsl/depthblur_fp.c build/release-darwin-arm64/renderergl2/glsl/dlight_fp.c build/release-darwin-arm64/renderergl2/glsl/dlight_vp.c build/release-darwin-arm64/renderergl2/glsl/down4x_vp.c build/release-darwin-arm64/renderergl2/glsl/bokeh_vp.c build/release-darwin-arm64/renderergl2/glsl/depthblur_vp.c build/release-darwin-arm64/renderergl2/glsl/calclevels4x_vp.c
make[2]: Verzeichnis „/nix/var/nix/builds/nix-build-ioq3-scion-unstable-2024-12-14.drv-0/source“ wird verlassen
make[1]: *** [Makefile:1639: targets] Fehler 2
make[1]: Verzeichnis „/nix/var/nix/builds/nix-build-ioq3-scion-unstable-2024-12-14.drv-0/source“ wird verlassen
make: *** [Makefile:1550: release] Fehler 2</pre>
</details>
</details>

dwt commented:
And another tidbit: the commercial Quake3 packages require non-free data files that need to be added to the nix store manually prior to building, so they will of course fail on the build server. That is expected.

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 390150`
Commit: `2a8e772fb192678b22c7afba4eb4851fa9a3d3a2`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 18 packages built:</summary>
  <ul>
    <li>collabora-online</li>
    <li>craftos-pc</li>
    <li>ioq3-scion</li>
    <li>ioquake3</li>
    <li>multipass</li>
    <li>mumble</li>
    <li>mumble_overlay</li>
    <li>murmur</li>
    <li>poco</li>
    <li>poco.dev</li>
    <li>pothos</li>
    <li>projectm-sdl-cpp</li>
    <li>quake3arena</li>
    <li>quake3arena-hires</li>
    <li>quake3demo</li>
    <li>quake3demo-hires</li>
    <li>sanjuuni</li>
    <li>vscode-extensions.jackmacwindows.craftos-pc</li>
  </ul>
</details>

GaetanLepage reviewed (approved):
All good, thanks!
